frontend/pkg/frontend/frontend.go

```diff
@@ -431,6 +509,227 @@ func (f *Frontend) ArmResourceCreateOrUpdate(writer http.ResponseWriter, request
 	}
 }
 
+// clusterResourceIdFromNodePoolResourceID strips the nodepool part of the full resource ID
+// returning the full cluster resourceID.
+// given a full resource ID like
+// /subscriptions/00000000-0000-0000-0000-000000000000/resourcegroups/dev-test-rg/providers/microsoft.redhatopenshift/hcpopenshiftclusters/aldo-cluster/nodepools/aldo-nodepool
+// it returns the resourceID of the cluster
+// /subscriptions/00000000-0000-0000-0000-000000000000/resourcegroups/dev-test-rg/providers/microsoft.redhatopenshift/hcpopenshiftclusters/aldo-cluster
+func clusterResourceIdFromNodePoolResourceID(resourceID string) string {
```

Comment:
Use https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azcore/arm#ParseResourceID to parse resource IDs.

Reply:
Sure. Right. Thanks!